### PR TITLE
dev-python/thonny: new package, add 4.0.1

### PR DESCRIPTION
**Pull request by corvus1**
A popular IDE for beginner programmers in python. Really comparable to the arduino platmorm, but more for Raspberry Pi Pico. It has quite the following. And yet, it is nowhere to be found in gentoo, not even in the overlays. I had to cobble together an ebuild, and stick it into bugzilla of all places. If a user wants it, they'd feel like they have to go dumpster diving for an ebuild. :-) But feel free to consider this PR a sad joke. The quality of the ebuild is not the greatest, sorry, but it is short and it works. Thanks for checking out. 

Closes: https://bugs.gentoo.org/666802
Signed-off-by: Michael Corvinus <voron1@gmail.com>

PS.
Oh, and check out that linked bug. 
The number and the date on it. :-)

### CODE REVIEW

**Comment by scardracs** [comment]
You forgot to make the metadata.xml

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+PYTHON_COMPAT=( python3_{8..11} )
+PYTHON_REQ_USE="tk"
+
+inherit distutils-r1 multiprocessing xdg-utils desktop
+
+DESCRIPTION="Thonny is a Python IDE meant for learning programming."
+HOMEPAGE="
+	https://thonny.org/
+	https://github.com/thonny/thonny
+"
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:16]: `HOMEPAGE="https://thonny.org/ https://github.com/thonny/thonny"`

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+SRC_URI="
+	https://github.com/thonny/thonny/archive/refs/tags/v${PV}.tar.gz -> ${PN}-${PV}.tar.gz
+"
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:19]: `SRC_URI="https://github.com/thonny/thonny/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"`

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		dev-python/setuptools[${PYTHON_USEDEP}]
+		dev-python/pyserial[${PYTHON_USEDEP}]
+		dev-python/pylint[${PYTHON_USEDEP}]
+		dev-python/docutils[${PYTHON_USEDEP}]
+		dev-python/mypy[${PYTHON_USEDEP}]
+		dev-python/asttokens[${PYTHON_USEDEP}]
+		dev-python/send2trash[${PYTHON_USEDEP}]
+	' 'python*'
+	)
+	"
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:37]: " without tabs

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+BDEPEND=""
+
+distutils_enable_tests pytest
+
+DEPEND=""
+
+RDEPEND+=${DEPEND}
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:45]: That's totally wrong, please fix it

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	https://thonny.org/
+	https://github.com/thonny/thonny
+"
+SRC_URI="
+	https://github.com/thonny/thonny/archive/refs/tags/v${PV}.tar.gz -> ${PN}-${PV}.tar.gz
+"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:23]: Only amd64 as I don't think you have tested it per-architecture

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+RDEPEND+=${DEPEND}
+
+src_prepare() {
+	default
+}
+
+src_install() {
+	distutils-r1_src_install
+	newicon packaging/icons/thonny-32x32.png thonny.png
+	domenu ${S}/packaging/linux/org.thonny.Thonny.desktop
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:54]: `domenu "${S}/packaging/linux/org.thonny.Thonny.desktop"`

**Review by scardracs — CHANGES_REQUESTED** [review]

File: dev-python/thonny/thonny-4.0.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_PEP517=setuptools
+PYTHON_COMPAT=( python3_{8..11} )
+PYTHON_REQ_USE="tk"
+
+inherit distutils-r1 multiprocessing xdg-utils desktop
```
Comment (scardracs) [inline dev-python/thonny/thonny-4.0.1.ebuild:10]: `inherit desktop distutils-r1 xdg-utils`

Comment (scardracs) [reply dev-python/thonny/thonny-4.0.1.ebuild:10]: They should always be alphabetically ordered

**Comment by scardracs** [comment]
Apart these errors I have to remember to you to put it into guru as manpower here is too little to maintain all these new ebuilds

**Comment by corvus1** [comment]
@scardracs Marсo, don't bother. I explained it in the first message. It is a sad joke. 
Thank you for your review and suggestions, you are of great help. With lazarus, I really couldn't do it without you. But here it's just hopeless.

**Comment by ajakk** [comment]
Please add to GURU as the first comment suggests: https://github.com/gentoo/gentoo/pull/29208#issuecomment-1399271478

**Pull request closed without merging**